arex-storage-web-api/src/main/java/com/arextest/storage/service/AgentWorkingService.java

```diff
@@ -153,6 +161,32 @@ public <T extends Mocker> byte[] queryMockResult(@NotNull T recordItem,
     return result;
   }
 
+  public byte[] queryMockers(String recordId, String[] fieldNames, String[] categoryList) {
+    List<RepositoryProvider<? extends Mocker>> repositoryProviderList = repositoryProviderFactory.getRepositoryProviderList();
+    Set<MockCategoryType> categoryTypes = repositoryProviderFactory.getCategoryTypes();
+
+    Set<MockCategoryType> searchCategoryTypes = ArrayUtils.isEmpty(categoryList) ?
+        categoryTypes :
+        Arrays.stream(categoryList)
+            .map(categoryName -> categoryTypes.stream()
+                .filter(categoryType -> categoryName.equals(categoryType.getName()))
+                .findFirst()
+                .orElse(null))
+            .filter(Objects::nonNull)
+            .collect(Collectors.toSet());
```

Comment:
The code for category type calculations is more appropriately placed in the `RepositoryProviderFactory` class

Reply:
The logic ends at com.arextest.storage.repository.RepositoryProviderFactory#getCategoryTypesByName